assets/src/application/ui/display/icons/svgs/withSize.tsx

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import classNames from 'classnames';
+import { withSizeProps, IconSize } from './types';
+import './style.scss';
+
+const withSize = <P extends withSizeProps>(WrappedComponent: React.ComponentType<P>) => {
+	type ISize = { size: IconSize };
+
+	const WithSize: React.FC<P & ISize> = ({ size, ...props }) => {
+		const className = classNames(props.className, size && `ee__icon--${size}`);
```

Comment:
can this not be baked right into our icon component?
or will that not work because those are dynamically generated?

I think of HOCs as enhancing a component with additional logic that doesn't belong in the component, whereas an icon's size seems like it should be a property of the icon itself, but there might be a technical reason why you have done this.

Reply:
yes, we can pass directly height and width:
```
<GlobalOutlined heght="1rem" noMargin width="1rem"/>
```
but I thought that having an universal approach is easier to maintain, rather than explicit settings